ReactiveXComponent/WebSocket/WebSocketPublisher.cs

```diff
@@ -1,35 +1,110 @@
+            var componentCode = _xcConfiguration.GetComponentCode(_component);
+            var topic = _xcConfiguration.GetPublisherTopic(_component, stateMachine);
+            var webSocketRequest = WebSocketMessageHelper.SerializeRequest(
+                    WebSocketCommand.Input,
+                    inputHeader,
+                    message,
+                    componentCode.ToString(),
+                    topic);
+
+            _webSocketClient.Send(webSocketRequest);
         }
 
         public void SendEvent(StateMachineRefHeader stateMachineRefHeader, object message, Visibility visibility = Visibility.Public)
         {
-            throw new NotImplementedException();
+            if (!_webSocketClient.IsOpen) return;
+
+            var inputHeader = CreateStateMachineRefWebSocketHeader(stateMachineRefHeader, message);
+            var componentCode = _xcConfiguration.GetComponentCode(_component);
+            var topic = _xcConfiguration.GetPublisherTopic(componentCode, stateMachineRefHeader.StateMachineCode);
+            var webSocketRequest = WebSocketMessageHelper.SerializeRequest(
+                    WebSocketCommand.Input,
+                    inputHeader,
+                    message,
+                    componentCode.ToString(),
+                    topic);
+
+            _webSocketClient.Send(webSocketRequest);
         }
 
         public List<MessageEventArgs> GetSnapshot(string stateMachine, int timeout = 10000)
         {
-            throw new NotImplementedException();
+            return _webSocketSnapshotManager.GetSnapshot(stateMachine, timeout);
         }
 
         public void GetSnapshotAsync(string stateMachine, Action<List<MessageEventArgs>> onSnapshotReceived)
         {
-            throw new NotImplementedException();
+            _webSocketSnapshotManager.GetSnapshotAsync(stateMachine, onSnapshotReceived);
         }
+        #endregion
 
+        private WebSocketEngineHeader CreateWebSocketHeader(string stateMachine, object message, Visibility visibility = Visibility.Public)
+        {
+            var messageType = message?.GetType();
+            var webSocketEngineHeader = new WebSocketEngineHeader
+            {
+                ComponentCode = new Option<long>(_xcConfiguration.GetComponentCode(_component)),
+                StateMachineCode = new Option<long>(_xcConfiguration.GetStateMachineCode(_component, stateMachine)),
+                EventCode = _xcConfiguration.GetPublisherEventCode(messageType?.ToString()),
+                MessageType = new Option<string>(messageType?.ToString()),
+                PublishTopic = visibility == Visibility.Private ? new Option<string>(_privateCommunicationIdentifier) : null
+            };
+
+            return webSocketEngineHeader;
+        }
+
+        private WebSocketEngineHeader CreateStateMachineRefWebSocketHeader(StateMachineRefHeader smRefHeader, object message)

```

Comment:
Shouldn't this be renamed to something like _CreateWebSocketHeaderFromStateMachineRef_ ?